peridynamics/model.py

```diff
@@ -126,15 +170,80 @@ def __init__(self, mesh_file, horizon, critical_strain, elastic_modulus,
+            `is_displacement_boundary` returns a (3) list of the boundary types
+            in each cartesian direction.
+            A boundary type with an int value of None if the particle is not
+            on a displacement controlled boundary, a value of 1 if is is on a
+            boundary and loaded in the positive cartesian direction, and a
+            value of -1 if it is on the boundary and loaded in the negative
+            direction, and a value of 0 if it is not loaded.
+        :type is_displacement_boundary: function
+        :arg is_forces_boundary: As 'is_displacement_boundary' but applying to
+            force boundary conditions as opposed to displacement boundary
+            conditions.
+        :type is_forces_boundary: function
+        :arg is_tip: A function to determine if a node is to be measured for
+            its reaction force or displacement over time, and if it is, which
+            direction the measurements are made
+            (positive or negative cartesian direction). It has the form
+            is_tip(:class:`numpy.ndarray`). The argument is the initial
+            coordinates of a particle being simulated. `is_tip` returns a
+            (3) list of the measurement types in each cartesian direction.
+            A boundary type with an int value of None if the particle is not on
+            the `tip` to be measured, a value of 1 if is is on the `tip` and
+            to be measured.
+        :type is_tip: function
+        :arg density: An (nnodes, ) array of node density values, each
+            corresponding to a material
+        :type density: :class:`numpy.ndarray`
+        :arg bond_types: The bond_types for the model.
+            If `None` the bond_types at the time of construction of the
+            :class:`Model` object will be used. Default `None`.
+        :type bond_types: :class:`numpy.ndarray`
+        :arg stiffness_corrections: The stiffness_corrections for
+            the model. If `None` the stiffness_corrections at the time
+            of construction of the :class:`Model` object will be used. Default
+            `None`.
+        :type stiffness_corrections: :class:`numpy.ndarray`
+        :arg int precise_stiffness_correction: A switch variable. Set to 1:
+            Stiffness corrections are calculated more accurately using
+            actual nodal volumes. Set to 0: Stiffness corrections are calculate
+            using an average nodal volume. Set to None: All stiffness
+            corrections are set to 1.0, i.e. no stiffness correction is
+            applied.
 
         :raises DimensionalityError: when an invalid `dimensions` argument is
             provided.
         :raises FamilyError: when a node has no neighbours (other nodes it
             interacts with) in the initial state.
+
+        :returns: A new :class:`Model` object.
+        :rtype: Model
         """
+        if not isinstance(integrator, Integrator):
```

Comment:
This is probably not needed as it is not useful information (apart from debugging), but if it was then it would definately require a verbose flag, as it may be best to keep the command line as simple as possible. I have added a warning for when model arrays (such as family, volume, nlist etc.) are read from positional arguments instead of calculated.
I have also added a warning when calculating the model arrays and nnodes > 4000 (arbritrarily large value) as it will take a while to calculate these, especially 'connectivity' tuple and 'family' array, as they require nested cython loop over nnodes.